--- a/app/src/main/java/com/best/deskclock/timer/TimerFragment.java
+++ b/app/src/main/java/com/best/deskclock/timer/TimerFragment.java
@@ -142,9 +142,13 @@
         });

         RecyclerView keyboardCarousel = mCreateTimerView.findViewById(R.id.quick_timer_carousel);
-        keyboardCarousel.setAdapter(mQuickTimerAdapter);
+        if (keyboardCarousel != null) {
+            keyboardCarousel.setAdapter(mQuickTimerAdapter);
+        }
         RecyclerView spinnerCarousel = mCreateTimerSpinnerView.findViewById(R.id.quick_timer_carousel);
-        spinnerCarousel.setAdapter(mQuickTimerAdapter);
+        if (spinnerCarousel != null) {
+            spinnerCarousel.setAdapter(mQuickTimerAdapter);
+        }

         mQuickTimerRepository.getAllQuickTimers().observe(getViewLifecycleOwner(), quickTimers -> {
             mQuickTimerAdapter.setQuickTimers(quickTimers);
